ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_completion_service.proto

```diff
@@ -33,9 +33,10 @@ service CommandCompletionService {
   // Errors:
   // - ``UNAUTHENTICATED``: if the request does not include a valid access token
   // - ``PERMISSION_DENIED``: if the claims in the token are insufficient to perform a given operation
-  // - ``NOT_FOUND``: if the request does not include a valid ledger id or if the ledger has been pruned before ``begin``
+  // - ``NOT_FOUND``: if the request does not include a valid ledger id
+  // - ``FAILED_PRECONDITION``: if the ledger has been pruned after the subscription start offset
   // - ``INVALID_ARGUMENT``: if the payload is malformed or is missing required fields
-  // - ``OUT_OF_RANGE``: if the absolute offset is not before the end of the ledger
+  // - ``OUT_OF_RANGE``: if the absolute offset is after the end of the ledger
```

Comment:
Have we changed the semantics of this error? Please note that `not before` is not the same as `after`.

Reply:
The semantics haven't changed in this PR. This a correction to reflect the status quo actually. See https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiCommandCompletionService.scala#L55-L56 which eventually dispatches to https://github.com/digital-asset/daml/blob/main/ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/LedgerOffsetValidator.scala#L59-L63

Reply:
This mismatch was there for a while afaict, since the logic using `gt` is there from at least mid 2020

Reply:
:+1: